File: app/src/main/java/com/sattar/j/smsfake/SmsFakeApplication.java
package com.sattar.j.smsfake;

import android.app.Application;
import android.content.Context;

import androidx.annotation.Nullable;
import androidx.multidex.MultiDex;


public class SmsFakeApplication extends Application {
    public static final int NORMAL_FONT = 0;
    public static final int BOLD_FONT = 1;
    public static final int MEDIUM_FONT = 2;
    public static final int LIGHT_FONT = 3;
    public static final int ULTRA_LIGHT_FONT = 4;

    @Override
    public void onCreate() {
        super.onCreate();
    }

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        MultiDex.install(this);

    }

    public static String getFont(int type) {
        switch (type) {
            case SmsFakeApplication.LIGHT_FONT:
                return "font/IRANSansLight.ttf";
            case SmsFakeApplication.BOLD_FONT:
                return "font/IRANSansBold.ttf";
            case SmsFakeApplication.MEDIUM_FONT:
                return "font/IRANSansMedium.ttf";
            case SmsFakeApplication.ULTRA_LIGHT_FONT:
                return "font/IRANSansUltraLight.ttf";
            default:
                return "font/IRANSans.ttf";
        }
    }
}

File: app/src/main/java/com/sattar/j/smsfake/SmsFakeApplication.kt
package com.sattar.j.smsfake

import android.app.Application
import android.content.Context
import androidx.multidex.MultiDex
import com.sattar.j.smsfake.data.network.api.ApiClient
import com.sattar.j.smsfake.tools.db
import com.sattar.j.smsfake.tools.repository
import com.sattar.j.smsfake.tools.retrofit
import com.sattar.j.smsfake.tools.viewModels
import org.koin.android.ext.koin.androidContext
import org.koin.core.context.startKoin

class SmsFakeApplication : Application() {
    companion object {
        var appContext: Context? = null
        const val NORMAL_FONT = 0
        const val BOLD_FONT = 1
        const val MEDIUM_FONT = 2
        const val LIGHT_FONT = 3
        const val ULTRA_LIGHT_FONT = 4

        @JvmStatic
        fun getFont(type: Int): String {
            return when (type) {
                LIGHT_FONT -> "font/IRANSansLight.ttf"
                BOLD_FONT -> "font/IRANSansBold.ttf"
                MEDIUM_FONT -> "font/IRANSansMedium.ttf"
                ULTRA_LIGHT_FONT -> "font/IRANSansUltraLight.ttf"
                else -> "font/IRANSans.ttf"
            }
        }
    }

    override fun onCreate() {
        super.onCreate()
        appContext = applicationContext
        startKoin {
            // Android context
            androidContext(this@SmsFakeApplication)
            // modules
            modules(listOf(
                    viewModels,
                    retrofit(ApiClient.BASE_URL),
                    db,
                    repository))
        }
    }

    override fun attachBaseContext(base: Context) {
        super.attachBaseContext(base)
        MultiDex.install(this)
    }


}